apps/api/src/notifications/notifications.service.ts

```diff
@@ -492,4 +509,88 @@ export class MarketingNotificationsService {
 
     return minutesPassed <= period
   }
+
+  private generateMapFromMailList(emailList: string[], contacts: ContactsMap): void {
+    for (const email of emailList) {
+      const id = randomUUID()
+
+      contacts.set(email, {
+        id: id,
+        hash: this.generateHash(id),
+        registered: false,
+      })
+    }
+  }
+
+  private updateMailListMap(
+    regUser: Person[],
+    contacts: ContactsMap,
+    skipAfterDate: Date,
+    unregisteredConsent: UnregisteredNotificationConsent[],
+  ) {
+    for (const registeredUser of regUser) {
+      const createdAt = new Date(registeredUser.createdAt)
+
+      // Remove email if it belongs to user created after the change has been deployed, as they had already decided
+      // whether to give consent or not.
+      if (contacts.get(registeredUser.email as string) && createdAt > skipAfterDate) {
+        Logger.debug(`Removing email ${registeredUser.email} from list`)
```

Comment:
Can we add a bit more in the log:

removing email XXX from the list because of date

and same later below:

removing email XXX because the user don't want them.